js/adapt-contrib-spoor.js

```diff
@@ -111,12 +111,9 @@ define(function(require) {
 			}
 			if (event.isPass) {
 				_.defer(_.bind(this.checkTrackingCriteriaMet, this));
-			} else {
-				var onAssessmentFailure = this.data._reporting._onAssessmentFailure;
-				if (onAssessmentFailure !== "") {
-					this.persistSuspendData();
-					scormWrapper.setStatus(onAssessmentFailure);
-				}
+			} else if (this.data._tracking._requireAssessmentPassed) {
+				this.persistSuspendData();
+				scormWrapper.setStatus(this.data._reporting._onAssessmentFailure);
```

Comment:
That's not quite there yet. Firstly, that call to ```persistSuspendData``` seems redundant (my mistake I think) - but more importantly there's no check that _onAssessmentFailure is set
It could also benefit from an explanatory comment before the ```else if```, something along the lines of 'if _requireAssessmentPassed is false we don't want to set the status as this could override the completion status set from completing all the content